BatchPreAnalysis.py

```diff
@@ -0,0 +1,131 @@
+
+def get_matching_files(directory, target, exclusion_String):
+    """
+    Search in directory and its subdirectories for files with target in their name and specific file ending.
+
+    :param directory: all files in this directory & subdirectories are checked
+    :param target: displacement or p_Bleach
+    :param ending: File ending
+    :return: List of matching file paths
+    """
+    matching_files = []
+    for path, subdirs, files in os.walk(directory):
+        for name in files:
+            if target in name.lower():
+                if exclusion_String not in name.lower():
+                    matching_files.append(os.path.join(path, name))
+    return matching_files
+
+def read_areas(directory):
+    files = os.listdir(directory)
+    files = [file for file in files if file.endswith(".csv")]
+    file_areas = [int(pd.read_csv(directory + "\\" + file)["Area"]) for file in files]
+    return files, file_areas
+
+
+def write_log(files, file_areas, save_log):
+    f = open(save_log, "w+")
+    f.write("Calibration:\n")
+    f.write('"X:", 1, "Y:", 1, "Units:", "", "Gray Units:", ""\n')
+    f.write('"Image Name", "Area"\n')
+
+    for file, file_area in zip(files, file_areas):
+        if file[-9:-4] == "_size":
+            file = file[:-9] + ".csv"
+        f.write('"' + file + '"' + ", " + str(file_area) + '\n')
+
+    f.close()
+    print(save_log + " is saved.")
+
+
+def main(config_path):
+    dirs = []
+    config = configparser.ConfigParser()
+    config.sections()
+    config.read(config_path)
+    try:
+        if len([key for key in config["INPUT_DIRS"]]):
+            for key in config["INPUT_DIRS"]:
+                dirs.append(config["INPUT_DIRS"][key])
+        else:
+            raise IncorrectConfigException("No input directory defined in config.")
+    except KeyError:
+        raise IncorrectConfigException("Section INPUT_DIRS missing in config.")
+    try:
+        software = config["SOFTWARE"]["software"]
+        if software == ("ThunderSTORM" or "rapidSTORM"):
+            pass
+        else:
+            raise IncorrectConfigException("Invalid Software")
+    except KeyError:
+        raise IncorrectConfigException("Parameter software missing in config.")
+    try:
+        pixel_size = config["CAMERA"]["pixel_size"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter pixel_size missing in config.")
+    try:
+        pixel_per_row = config["CAMERA"]["pixel_per_row"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter pixel_per_row missing in config.")
+    try:
+        camera_integration_time = config["CAMERA"]["camera_integration_time"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter camera_integration_time missing in config.")
+    try:
+        background_size = config["CAMERA"]["background_size"]
+    except KeyError:
+        raise IncorrectConfigException("Parameter background_size missing in config.")
+
+
+    for dir in dirs:
```

Comment:
could you add a few inline comments here that describe what is happening in this block of code (line 102-122)?